geode-assembly/src/acceptanceTest/java/org/apache/geode/metrics/CacheGetsTimerTest.java

```diff
@@ -0,0 +1,521 @@
+
+    assertThat(allTimerValuesForRegion(replicateRegion))
+        .as("All timer values before destroying region " + replicateRegion.getName())
+        .hasSize(2);
+
+    replicateRegion.destroyRegion();
+
+    assertThat(allTimerValuesForRegion(replicateRegion))
+        .as("All timer values after destroying region " + replicateRegion.getName())
+        .isEmpty();
+  }
+
+  @Test
+  public void timersExistWithInitialValues_ifNoGetsPerformedOnPartitionRegion() throws IOException {
+    startWithTimeStatisticsEnabled();
+
+    assertThat(allTimerValuesForRegion(partitionRegion))
+        .as("All timer values for region " + partitionRegion.getName())
+        .hasSize(2)
+        .anyMatch(tv -> tv.result.equals("hit"))
+        .anyMatch(tv -> tv.result.equals("miss"))
+        .allMatch(tv -> tv.count == 0, "All timers have count of zero")
+        .allMatch(tv -> tv.totalTime == 0, "All timers have total time of zero");
+  }
+
+  @Test
+  public void allTimersRemoved_ifPartitionRegionDestroyed() throws IOException {
+    startWithTimeStatisticsEnabled();
+
+    assertThat(allTimerValuesForRegion(partitionRegion))
+        .as("All timer values before destroying region " + partitionRegion.getName())
+        .hasSize(2);
+
+    partitionRegion.destroyRegion();
+
+    assertThat(allTimerValuesForRegion(partitionRegion))
+        .as("All timer values after destroying region " + partitionRegion.getName())
+        .isEmpty();
+  }
+
+  @Test
+  public void timersRecordCountForReplicateRegion_ifTimeStatisticsDisabled() throws IOException {
+    startWithTimeStatisticsDisabled();
+
+    replicateRegion.put("existing-key", "existing-value");
+    replicateRegion.get("existing-key");
+    replicateRegion.get("missing-key");
+
+    assertThat(allTimerValuesForRegion(replicateRegion))
+        .as("All timer values for region " + replicateRegion.getName())
+        .hasSize(2)
+        .anyMatch(tv -> tv.result.equals("hit"))
+        .anyMatch(tv -> tv.result.equals("miss"))
+        .allMatch(tv -> tv.count == 1, "All timers have count of one")
+        .allMatch(tv -> tv.totalTime == 0, "All timers have total time of zero");
+  }
+
+  @Test
+  public void timersRecordCountForPartitionRegion_ifTimeStatisticsDisabled() throws IOException {
+    startWithTimeStatisticsDisabled();
+
+    partitionRegion.put("existing-key", "existing-value");
+    partitionRegion.get("existing-key");
+    partitionRegion.get("missing-key");
+
+    assertThat(allTimerValuesForRegion(partitionRegion))
+        .as("All timer values for region " + partitionRegion.getName())
+        .hasSize(2)
+        .anyMatch(tv -> tv.result.equals("hit"))
+        .anyMatch(tv -> tv.result.equals("miss"))
+        .allMatch(tv -> tv.count == 1, "All timers have count of one")
+        .allMatch(tv -> tv.totalTime == 0, "All timers have total time of zero");
+  }
+
+  @Test
+  public void timersDoNotRecord_ifNotAuthorized() throws IOException {
```

Comment:
I would give this a more general name, because the responsibility to refrain from recording is not about authorization specifically, but about exceptions in general. Authorization is merely one example of that.

Consider: `...ifGetThrowsException()`

Reply:
Addressed in latest commit